### ISSUE TITLE
geany 1.27 triggers a libc crash on ArchLinux sometimes

### ISSUE THREAD

**Issue by monojp** [geany/geany#1031]
Hi, I seem to have segfaults with geany 1.27 on Linux 4.5.3 with glibc 2.23. They happen very seldom but if they do I can trigger them by editing the same file the crash happend on a certain position. Is this a geany problem or should it be reported somewhere else? I managed to get a very simple backtrace via gdb.
```
[ 4442.960978] geany[1321]: segfault at 0 ip 00007f094992d88e sp 00007fff8e1764f8 error 4 in libc-2.23.so[7f09498a0000+197000
```
```
$ gdb geany 
GNU gdb (GDB) 7.11
Copyright (C) 2016 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.  Type "show copying"
and "show warranty" for details.
This GDB was configured as "x86_64-pc-linux-gnu".
Type "show configuration" for configuration details.
For bug reporting instructions, please see:
<http://www.gnu.org/software/gdb/bugs/>.
Find the GDB manual and other documentation resources online at:
---Type <return> to continue, or q <return> to quit---
<http://www.gnu.org/software/gdb/documentation/>.
For help, type "help".
Type "apropos word" to search for commands related to "word"...
Reading symbols from geany...(no debugging symbols found)...done.
(gdb) start
Function "main" not defined.
Make breakpoint pending on future shared library load? (y or [n]) 
Starting program: /usr/bin/geany 
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffeeb96700 (LWP 18778)]
[New Thread 0x7fffee395700 (LWP 18779)]
[New Thread 0x7fffedb94700 (LWP 18780)]
[New Thread 0x7fffed393700 (LWP 18782)]
[Thread 0x7fffedb94700 (LWP 18780) exited]
[New Thread 0x7fffedb94700 (LWP 18783)]
[New Thread 0x7fffde233700 (LWP 18784)]
[New Thread 0x7fffdd056700 (LWP 18785)]
[New Thread 0x7fffdc855700 (LWP 18786)]
[New Thread 0x7fffc7fff700 (LWP 18787)]
[Thread 0x7fffdd056700 (LWP 18785) exited]
[Thread 0x7fffc7fff700 (LWP 18787) exited]
[Thread 0x7fffdc855700 (LWP 18786) exited]
[Thread 0x7fffee395700 (LWP 18779) exited]
[Thread 0x7fffedb94700 (LWP 18783) exited]
[New Thread 0x7fffedb94700 (LWP 18788)]
[Thread 0x7fffedb94700 (LWP 18788) exited]
[New Thread 0x7fffedb94700 (LWP 18803)]
[New Thread 0x7fffc7fff700 (LWP 18804)]
[New Thread 0x7fffdc855700 (LWP 18805)]
[New Thread 0x7fffee395700 (LWP 18806)]
[New Thread 0x7fffdd056700 (LWP 18807)]
[Thread 0x7fffde233700 (LWP 18784) exited]
[Thread 0x7fffc7fff700 (LWP 18804) exited]
[Thread 0x7fffee395700 (LWP 18806) exited]
[Thread 0x7fffdd056700 (LWP 18807) exited]
[Thread 0x7fffdc855700 (LWP 18805) exited]
[New Thread 0x7fffdd056700 (LWP 18808)]
[New Thread 0x7fffdc855700 (LWP 18809)]
[New Thread 0x7fffee395700 (LWP 18810)]
[New Thread 0x7fffc7fff700 (LWP 18811)]
[New Thread 0x7fffde233700 (LWP 18812)]
[New Thread 0x7fffc77fe700 (LWP 18813)]
[New Thread 0x7fffc6ffd700 (LWP 18814)]
[New Thread 0x7fffc67fc700 (LWP 18815)]
[New Thread 0x7fffc5ffb700 (LWP 18816)]
[Thread 0x7fffc67fc700 (LWP 18815) exited]
[Thread 0x7fffc7fff700 (LWP 18811) exited]
[Thread 0x7fffdc855700 (LWP 18809) exited]
[Thread 0x7fffedb94700 (LWP 18803) exited]
[Thread 0x7fffc6ffd700 (LWP 18814) exited]
[Thread 0x7fffc77fe700 (LWP 18813) exited]
[Thread 0x7fffee395700 (LWP 18810) exited]
[Thread 0x7fffdd056700 (LWP 18808) exited]
[Thread 0x7fffde233700 (LWP 18812) exited]
[New Thread 0x7fffde233700 (LWP 18817)]
[Thread 0x7fffde233700 (LWP 18817) exited]
[New Thread 0x7fffde233700 (LWP 18832)]

(geany:18774): Geany-CRITICAL **: sci_get_contents_range: assertion 'start < end' failed

Thread 1 "geany" received signal SIGSEGV, Segmentation fault.
0x00007ffff740f88e in __strcmp_sse2_unaligned () from /usr/lib/libc.so.6
(gdb) backtrace
#0  0x00007ffff740f88e in __strcmp_sse2_unaligned () from /usr/lib/libc.so.6
#1  0x00007fffde4f84e1 in ?? () from /usr/lib/geany/pairtaghighlighter.so
#2  0x00007ffff79a290c in ?? () from /usr/lib/libgeany.so.0
#3  0x00007ffff51cafa5 in g_closure_invoke () from /usr/lib/libgobject-2.0.so.0
#4  0x00007ffff51dcff1 in ?? () from /usr/lib/libgobject-2.0.so.0
#5  0x00007ffff51e5829 in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
#6  0x00007ffff51e6585 in g_signal_emit_by_name () from /usr/lib/libgobject-2.0.so.0
#7  0x00007ffff7994d9f in ?? () from /usr/lib/libgeany.so.0
#8  0x00007ffff51cafa5 in g_closure_invoke () from /usr/lib/libgobject-2.0.so.0
#9  0x00007ffff51dcff1 in ?? () from /usr/lib/libgobject-2.0.so.0
#10 0x00007ffff51e5d8c in g_signal_emit_valist () from /usr/lib/libgobject-2.0.so.0
---Type <return> to continue, or q <return> to quit---
#11 0x00007ffff51e60bf in g_signal_emit () from /usr/lib/libgobject-2.0.so.0
#12 0x00007ffff79f055f in ?? () from /usr/lib/libgeany.so.0
#13 0x00007ffff7a1579b in ?? () from /usr/lib/libgeany.so.0
#14 0x00007ffff7a1cc7a in ?? () from /usr/lib/libgeany.so.0
#15 0x00007ffff79f0065 in ?? () from /usr/lib/libgeany.so.0
#16 0x00007ffff4aeedba in g_main_context_dispatch () from /usr/lib/libglib-2.0.so.0
#17 0x00007ffff4aef160 in ?? () from /usr/lib/libglib-2.0.so.0
#18 0x00007ffff4aef482 in g_main_loop_run () from /usr/lib/libglib-2.0.so.0
#19 0x00007ffff6e6f4e7 in gtk_main () from /usr/lib/libgtk-x11-2.0.so.0
#20 0x00007ffff79b2b07 in main_lib () from /usr/lib/libgeany.so.0
#21 0x00007ffff73a2741 in __libc_start_main () from /usr/lib/libc.so.6
#22 0x00000000004005e9 in ?? ()
```

Sorry if this is geany unrelated.

**Comment by eht16** [comment]
The backtrace reads like a duplicate of https://github.com/geany/geany/issues/1016.

**Comment by monojp** [comment]
Yeah you're right, I didn't see that bug. It looks like it comes from the pairtaghighlighter plugin. Thank your for your fast response!
